File: TTObjectClasses/Tıbbi Süreç/Kayıt - Kabul Modülü/Requirements/PatientAddressRequirements.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TTUtils.RequirementManager;

namespace TTObjectClasses.Tıbbi_Süreç.Kayıt___Kabul_Modülü.Requirements
{
  
    public class PatientAddressRequirements : IRequirement
    {
        PatientIdentityAndAddress patientAddress = null;
        public PatientAddressRequirements(PatientIdentityAndAddress patientAddressParam)
        {
            patientAddress = patientAddressParam;

        }

        public RequirementResultCode ExecuteRequirement()
        {
            RequirementResultCode requirementResultCode = new RequirementResultCode();
            requirementResultCode.resultCode = 0;
            requirementResultCode.resultMessage = "Success";

            if (patientAddress == null)
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25776", "Hasta adresi tanımlı değildir.");
                return requirementResultCode;
            }
            if (patientAddress.HomeAddress == null || patientAddress.HomeAddress == "")
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25621", "Ev adresi alanı boş bırakılamaz.");
                return requirementResultCode;
            }
            if (patientAddress.MobilePhone == null || patientAddress.MobilePhone == "")
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25357", "Cep telefonu alanı boş bırakılamaz.");
                return requirementResultCode;
            }
            return requirementResultCode;
        }
    }

}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Kayit_Kabul_Modulu/Requirements/PatientAddressRequirements.ts
//$90DCFB62
import { RequirementResultCode } from "Fw/Requirements/RequirementResultCode";
import { IRequirements } from "Fw/Requirements/IRequirements";
import { PatientAdmissionFormViewModel } from "../PatientAdmissionFormViewModel";

export class PatientAddressRequirements implements IRequirements {

    public PA_Address: PatientAdmissionFormViewModel;

    public requiredColor: string = "#f3d7d7";
    public PatientAddressUIElements: Map<string, any>;

    constructor(patientAddressParam: PatientAdmissionFormViewModel, controls: Map<string, any>) {
        this.PA_Address = patientAddressParam;
        this.PatientAddressUIElements = controls;
    }
    ExecuteRequirementsWithApproval(): Promise<RequirementResultCode> {
        throw new Error("Method not implemented.");
    }
    public ExecuteRequirements(): RequirementResultCode {

        let requirementResultCode: RequirementResultCode = new RequirementResultCode();
        requirementResultCode.resultCode = 0;
        requirementResultCode.resultMessage = "Success";

        let UIComponent = null;

        if (this.PA_Address == null) {
            requirementResultCode.resultCode = 1;
            requirementResultCode.resultMessage = i18n("M15139", "Hasta adresi tanımlı değildir.");
            return requirementResultCode;
        }

        if (this.PA_Address.tempHomeAddress == null || this.PA_Address.tempHomeAddress == "") {
            requirementResultCode.resultCode = 1;
            requirementResultCode.resultMessage = i18n("M13974", "Ev adresi alanı boş bırakılamaz.");
            UIComponent = this.PatientAddressUIElements.get("HomeAddress");
            if (UIComponent != null)
                UIComponent.BackColor = this.requiredColor;
        }

        if (this.PA_Address.tempMobilePhone == null || this.PA_Address.tempMobilePhone.replace(/\s/g, "") == "") {
            requirementResultCode.resultCode = 1;
            requirementResultCode.resultMessage = i18n("M12199", "Cep telefonu alanı boş bırakılamaz.");
            UIComponent = this.PatientAddressUIElements.get("MobilePhone");
            if (UIComponent != null)
                UIComponent.BackColor = this.requiredColor;
        }
        else if (this.PA_Address.tempMobilePhone.replace(/\s/g, "").length < 10) {
            requirementResultCode.resultCode = 1;
            requirementResultCode.resultMessage = i18n("M12199", "Cep telefonu alanı uzaunluğu 10 karakter olmalıdır.");
            UIComponent = this.PatientAddressUIElements.get("MobilePhone");
            if (UIComponent != null)
                UIComponent.BackColor = this.requiredColor;
        }

        return requirementResultCode;
    }

}
